lpwan-backend/src/main/java/com/cumulocity/lpwan/lns/connection/model/LpwanDeviceFilter.java

```diff
@@ -4,16 +4,14 @@
 import com.cumulocity.sdk.client.inventory.InventoryFilter;
 
 public class LpwanDeviceFilter extends InventoryFilter {
-
     @ParamSource
     private String query;
 
-    private LpwanDeviceFilter(String filterCriteria, String filterCriteriaValue) {
-        query = "$filter=c8y_LpwanDevice." + filterCriteria + " eq " +
-                String.format("'%s'", filterCriteriaValue);
+    private LpwanDeviceFilter(String serviceProvider, String lnsConnectionName) {
+        this.query = String.format("$filter=c8y_LpwanDevice.serviceProvider eq '%s' and c8y_LpwanDevice.lnsConnectionName eq '%s'", serviceProvider, lnsConnectionName);
```

Comment:
The reason for writing the query in this form instead of using a variable and then doing String.format:

When we do inventoryApi.getManagedObjectsByFilter, we call filter.getQueryParams. This iterates through the declared fields. 

While doing so, a call is made to getParamName in this snippet:

String value = (String) safelyGetFieldValue(field, this);
            if (value != null) {
                params.put(getParamName(field), encode(value));
            }

getParamName checks if the field has ParamSource annotation. Since the private static field used doesnt have the ParamSource annotation, field.getAnnotation(ParamSource.class) will throw NullPointerException